src/event-tracker.ts

```diff
@@ -94,7 +93,20 @@ export class EventTracker {
 			// Call this function at maximum once every minute
 			return _.throttle(
 				properties => {
-					this.client.track(event, properties);
+					if (this.client != null) {
+						if (!properties.uuid) {
+							// In some weird cases we see events that seem to be submitted by the supervisor
+							// but not containing any default properties.
+							// This code is supposed to clarify if it actually happens because of the supervisor.
+							const availableData = _.merge({}, this.defaultProperties, {
+								details: 'uuid is missing on event properties',
+							});
+							this.client.track(`Problematic ${event}`, availableData);
```

Comment:
@gelbal this way we'll at least have a prove it's coming from supervisor if it's reproduced in newer versions

@CameronDiver it's related to the fact we see "Supervisor start" events without uuid reported. It may be coming from some old supervisor versions. In our current version, I could find any real race that would lead to firing an event without default properties being set. But just to make sure, I'm adding this. What do you think?

Reply:
Sounds good Roman. Talking in general, for such debugging cases, is it possible to send events only to Sentry and not to Mixpanel/Amplitude?

If possible, I'd prefer not to pollute the Mixpanel/Amplitude prod schema for debug reasons. Then of course you could argue that the schema is already "polluted" with an unexpected event : )

Reply:
The problem here is that Sentry is not integrated into the supervisor. Hence, we can only write to the logs or send via mixpanel client.

Reply:
I've not tested this at all, but there strictly is some time when the supervisor starts up and it doesn't have a UUID, e.g. before provision. Is this not the event you're investigating?

In general, I'm for these kinds of changes, but I'd like to understand a little more about this stray event you're referencing. Furthermore, I'm wondering if the supervisor start event is even useful, but maybe that is a bigger conversation.

Reply:
An unprovisioned device is a good point and must explain this. Thanks!

Reply:
Removed this code.